src/crd.rs

```diff
@@ -74,8 +82,22 @@ where
         Ok(())
     } else {
         info!("CRD not detected in Kubernetes. Attempting to create it.");
-        create::<T>(client).await
-        // TODO: Maybe retry?
+
+        let crd_created = async {
+            loop {
+                if let Ok(res) = create::<T>(client.clone()).await {
+                    break res;
+                }
+                tokio::time::sleep(Duration::from_millis(100)).await;
+            }
+            wait_ready::<T>(client.clone()).await?;
+            Ok(())
+        };
+        if let Some(timeout) = timeout {
```

Comment:
Would it make sense to extract this to a `wait_created_with_timeout` method next to the `wait_created` one? This seems like a super useful pattern to me and it's not limited to CRDs

Reply:
First, the naming is wrong and it should be aligned with `wait_created` on Client. So I renamed it.

`wait_created` and its documentation clearly says the resources is created in Kubernetes and in some context (CRDs) recognized by Kubernetes APIs. Nothing more. There are no implications about resource's state. If that was a pod, it could be created, yet the underlying container could be crashed. I **avoided** the word `ready`, because readiness has different meaning in Kubernetes, not to say the meaning changes with context. E.g. a ready pod is generally understood to be a pod that's has `status.Ready=true`.

Reply:
This could be one generic method. But hey, it's already there. The `Option<Duration>` passed as argument into this method is only used in `tokio::time::timeout`. What if we just let the callers handle the timeout themselves ? 

Instead of calling the function with Duration,

```rust
    ensure_crd_created::<TestCrd>(client.clone(), Some(Duration::from_secs(10)))
        .await
```

the function signature is simplified and the timeout is delegated to that dedicated function, which to me is `tokio::time::timeout` and we'd just duplicate it's functionality. It looks like this.

```rust
    tokio::time::timeout(
        Duration::from_secs(30),
        ensure_crd_created::<TestCrd>(client.clone()),
    )
    .await
```
Also, the code is cleaner and it clearly states there is a timeout.

Reply:
I did the above mentioned change for you to see the effects. It is contained in a completely separate commit https://github.com/stackabletech/operator-rs/pull/79/commits/3b2e5b078332a11df2adf67152fcae0803310442 - we can always roll this one back.

Reply:
Makes sense to me! We can always add something later.